pixeltable/catalog/catalog.py

```diff
@@ -409,16 +412,45 @@ def begin_xact(
                         self._clear_tv_cache(handle.id, handle.effective_version)
                 self._modified_tvs.clear()
 
-    def raise_from_sql_exc(
-        self, e: sql_exc.StatementError, tbl_id: UUID | None, tbl: TableVersionHandle | None, convert_db_excs: bool
+    @contextmanager
+    def __begin_guarded_xact(self, *, for_write: bool) -> Iterator[sql.Connection]:
```

Comment:
Missing comment. When do I use this?

Reply:
Never - that's why it's marked private. It should only be called by begin_xact.